### PR TITLE
Add FIFO and LIFO approaches to InventoryStrategy project

### PR DESCRIPTION
**Pull request by rozhaaan**

### CODE REVIEW

File: sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        _products = products;
+    }
+
+    public Product Current => _products[_currentProductIndex];
+
+    object IEnumerator.Current => Current;
+
+    public void Dispose()
+    {
```
Comment (thisisnabi) [inline sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs:21]: Clear does not destroy the object.
The correct way is to set the list to null.

`_products=null;`

File: sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        _products.Clear();
+    }
+
+    public bool MoveNext()
+    {
+        while (_currentProductIndex++ < _products.Count)
```
Comment (thisisnabi) [inline sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs:27]: Move the index increment to another statement, this will simplify your code.

**Review by thisisnabi — APPROVED** [review]
I have given some things for your information, but the code is acceptable.

File: sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs (hunk 2 above)
Comment (rozhaaan) [reply sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs:27]: Something like this?

_index ++;
while (_index < _products.Count) {...}

File: sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs (hunk 1 above)
Comment (rozhaaan) [reply sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs:21]: I'll keep in mind, thank you

File: sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs (hunk 2 above)
Comment (thisisnabi) [reply sources/Morvarid Arab/ConsoleApp/Enumerators/FIFOEnumerator.cs:27]:
```csharp
Index++;
return index < countOfProduct;

```

**Pull request merged**